Python/compile.c

```diff
@@ -6883,6 +6888,22 @@ optimize_basic_block(basicblock *bb, PyObject *consts)
                             bb->b_instr[i+1].i_opcode = NOP;
                         }
                         break;
+                    case BINARY_ADD:
+                        cnt = PyList_GET_ITEM(consts, oparg);
+                        if (PyLong_CheckExact(cnt) && inst->i_lineno == bb->b_instr[i+1].i_lineno) {
+                            int ovf = 0;
+                            long val = PyLong_AsLongAndOverflow(cnt, &ovf);
+                            // TODO: What about larger values?
```

Comment:
Anything beyond a single EXTENDED_ARG is going to be a loss. For one EXTENDED_ARG, there is no advantage for dispatching, but we still gain from specialization.
Having said that, I suspect in practice it will be irrelevant as most constant ints will be small.

Reply:
Okay, I'll leave this comment in and won't take any action now. We can reconsider this in the future.